endpoints/data_objects/src/main.cpp

```diff
@@ -2305,6 +2308,134 @@ namespace
+		                                       client_info,
+		                                       _sess_ptr,
+		                                       _req = std::move(_req),
+		                                       _args = std::move(_args)] {
+			logging::info(*_sess_ptr, "{}: client_info.username = [{}]", fn, client_info.username);
+
+			http::response<http::string_body> res{http::status::ok, _req.version()};
+			res.set(http::field::server, irods::http::version::server_name);
+			res.set(http::field::content_type, "application/json");
+			res.keep_alive(_req.keep_alive());
+
+			try {
+				const auto lpath_iter = _args.find("lpath");
+				if (lpath_iter == std::end(_args)) {
+					logging::error(*_sess_ptr, "{}: Missing [lpath] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+				}
+
+				const auto size_iter = _args.find("size");
+				if (lpath_iter == std::end(_args)) {
+					logging::error(*_sess_ptr, "{}: Missing [size] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+				}
+
+				DataObjInp input{};
+				lpath_iter->second.copy(input.objPath, sizeof(DataObjInp::objPath) - 1);
+
+				try {
+					input.dataSize = std::stoll(size_iter->second);
+				}
+				catch (const std::exception& e) {
+					logging::error(
+						*_sess_ptr, "{}: Could not convert size [{}] into an integer.", fn, size_iter->second);
+					return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+				}
+
+				if (const auto iter = _args.find("resource"); iter != std::end(_args)) {
+#if 0
+					if (_args.find("replica-number") != std::end(_args)) {
+						logging::error(
+							*_sess_ptr,
+							"{}: [resource], [replica-number], and [resource-hierarchy] parameters are incompatible.",
+							fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
+#endif
+
+					addKeyVal(&input.condInput, RESC_NAME_KW, iter->second.c_str());
+				}
+
+				if (const auto iter = _args.find("replica-number"); iter != std::end(_args)) {
+#if 0
+					if (_args.find("resource") != std::end(_args)) {
+						logging::error(
+							*_sess_ptr,
+							"{}: [resource], [replica-number], and [resource-hierarchy] parameters are incompatible.",
+							fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
+#endif
+
+					addKeyVal(&input.condInput, REPL_NUM_KW, iter->second.c_str());
+				}
+
+				if (const auto iter = _args.find("resource-hierarchy"); iter != std::end(_args)) {
+#if 0
+					if (_args.find("resource") != std::end(_args)) {
+						logging::error(
+							*_sess_ptr,
+							"{}: [resource], [replica-number], and [resource-hierarchy] parameters are incompatible.",
+							fn);
+						return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+					}
+#endif
+
+					addKeyVal(&input.condInput, RESC_HIER_STR_KW, iter->second.c_str());
+				}
```

Comment:
After some discussion, we've decided to not expose the resource hierarchy option.